src/common/java/net/minecraftforge/gradle/common/legacy/LegacyExtension.java

```diff
@@ -0,0 +1,141 @@
+                t.getZip().fileProvider(project.provider(() -> {
+                    // create maven dependency for mappings
+                    String MAPPING_DEP = "net.minecraft:mappings_{CHANNEL}:{VERSION}@zip";
+                    String coordinates = MAPPING_DEP.replace("{CHANNEL}", minecraft.getMappingChannel().get()).replace("{VERSION}", minecraft.getMappingVersion().get());
+
+                    Dependency dep = project.getDependencies().create(coordinates);
+                    // download and cache the mappings
+                    return project.getConfigurations().detachedConfiguration(dep).getSingleFile();
+                }));
+                // mappings are extracted to <root>/build/<taskName> by default
+                t.getOutput().convention(project.getLayout().getBuildDirectory().dir(t.getName()));
+            });
+            final File csvDir = extractMappingsTask.getOutput().get().getAsFile();
+
+            // attach the csvDir location property to each run configuration
+            minecraft.getRuns().configureEach(run -> run.property("net.minecraftforge.gradle.GradleStart.csvDir", csvDir));
+
+            // execute extractMappings before each run task
+            project.getTasks().getByName("prepareRuns").dependsOn(extractMappingsTask);
+        }
+    }
+
+    public LegacyExtension(Project project) {
+        Provider<Boolean> isLegacy = project.provider(() -> {
+            final MinecraftVersion version = MinecraftVersion.from((String) (project.getParent() != null ? project.getParent() : project).getExtensions().getExtraProperties().get("MC_VERSION"));
+            
+            // enable patches by default if version is below FG 3
+            return version.compareTo(FG3) < 0;
+        });
+        
+        getFixClasspath().convention(isLegacy);
+        getExtractMappings().convention(isLegacy);
+    }
+
+    /**
+     * fixClassPath;
+     *  FG2 Userdev puts all classes and resources into a single jar file for FML to consume.
+     *  FG3+ puts classes and resources into separate folders, which breaks on older versions.
+     *  We replicate the FG2 behavior by replacing these folders by the jar artifact on the runtime classpath.
+     *
+     * Takes a boolean - true for apply fix, false for no fix.
+     */
+    public abstract Property<Boolean> getFixClasspath();
+
+    /**
+     * extractMappings;
+     * FG2 GradleStart exposes a <code>net.minecraftforge.gradle.GradleStart.csvDir</code> property
+     * that points to a directory containing CSV mappings files. This is used by LegacyDev to remap dependencies' AT modifiers.
+     * We replicate the FG2 behavior by extracting the mappings to a folder in the build directory
+     * and setting the property to point to it.
+     * <p>
+     * Takes a boolean - true for apply fix, false for no fix.
+     */
+    public abstract Property<Boolean> getExtractMappings();
```

Comment:
Remove the public properties.
The PR for this states that this object is not supposed to be a part of the public API.

If this is part of the public API, then we should discuss that.

Reply:
Do you expect these properti 's tok be set by a user?

Reply:
Let me state this differently, is there any **legacy** version of Minecraft for which this extension is designed which has either one set to false?